src/expr.rs

```diff
@@ -1134,10 +1134,17 @@ impl<'a> ControlFlow<'a> {
         // 1 = space after keyword.
         let offset = self.keyword.len() + label_string.len() + 1;
 
+        let brace_overhead =
+            if context.config.control_brace_style() != ControlBraceStyle::AlwaysNextLine {
+                // 2 = ` {`
+                2
+            } else {
+                0
+            };
         let pat_expr_string = match self.cond {
             Some(cond) => {
                 let cond_shape = match context.config.control_style() {
-                    Style::Legacy => try_opt!(constr_shape.shrink_left(offset)),
+                    Style::Legacy => try_opt!(constr_shape.shrink_left(offset + brace_overhead)),
```

Comment:
Rather than adding `brace_overhead` here, we could remove `context.config.control_style() == Style::Rfc &&`
from [this line](https://github.com/spinda/rustfmt/blob/20e3485d6c5f06e0221aa8730354603e25618cc3/src/expr.rs#L1168) to fix the issue.